### PR TITLE
ISPN-11803 Move Optimistic Transaction versions to PrivateMetadata

### PR DESCRIPTION
**Pull request by pruivo**
https://issues.redhat.com/browse/ISPN-11803

### CODE REVIEW

File: core/src/main/java/org/infinispan/metadata/impl/PrivateMetadata.java [hunk 1]
```diff
@@ -74,29 +87,75 @@ public IracMetadata iracMetadata() {
...
+            Objects.equals(entryVersion, metadata.entryVersion);
+   }
+
+   @Override
+   public int hashCode() {
+      return Objects.hash(iracMetadata, entryVersion);
+   }
+
+   @ProtoField(number = 2)
+   public NumericVersion getNVersion() {
```
Comment (wburns) [inline core/src/main/java/org/infinispan/metadata/impl/PrivateMetadata.java:131]:
```suggestion
   public NumericVersion getNumericVersion() {
```

File: core/src/main/java/org/infinispan/metadata/impl/PrivateMetadata.java [hunk 2]
```diff
@@ -74,29 +87,75 @@ public IracMetadata iracMetadata() {
...
+      return entryVersion instanceof NumericVersion ? (NumericVersion) entryVersion : null;
+   }
+
+   @ProtoField(number = 3)
+   public SimpleClusteredVersion getCVersion() {
```
Comment (wburns) [inline core/src/main/java/org/infinispan/metadata/impl/PrivateMetadata.java:136]:
```suggestion
   public SimpleClusteredVersion getClusteredVersion() {
```

**Comment by pruivo** [comment]
updated :)

**Comment by wburns** [comment]
Integrated into master, thanks @pruivo !

**Pull request merged**